src/main/scala/beam/agentsim/agents/modalBehaviors/DrivesVehicle.scala

```diff
@@ -0,0 +1,186 @@
+      _currentTick = Some(tick)
+      passengerSchedule.schedule.get(completedLeg) match {
+        case Some(manifest) =>
+          _awaitingAlightConfirmation ++= manifest.alighters
+          manifest.riders.foreach(passenger => beamServices.vehicleRefs(passenger) ! NotifyLegEnd)
+          stay()
+        case None =>
+          log.error(s"Driver ${id} did not find a manifest for BeamLeg ${_currentLeg}")
+          goto(BeamAgent.Error)
+      }
+    case Event(ReservationRequestWithVehicle(req, vehicleData: VehicleData), _) =>
+      val proxyVehicleAgent = sender()
+      require(passengerSchedule.schedule.nonEmpty, "Driver needs to init list of stops")
+      //      val response: ReservationResponse = handleVehicleReservation(req, vehicleData, proxyVehicleAgent)
+      //        req.passenger ! response
+      stay()
+    case Event(AlightingConfirmation(vehicleId), agentInfo) =>
+      _awaitingAlightConfirmation -= vehicleId
+      if (_awaitingAlightConfirmation.isEmpty) {
+        processNextLegOrCompleteMission()
+      } else {
+        stay()
+      }
+  }
+  chainedWhen(Waiting) {
+    case Event(BecomeDriverSuccess(newPassengerSchedule), info) =>
+      newPassengerSchedule match {
+        case Some(passSched) =>
+          passengerSchedule = passSched
+        case None =>
+          passengerSchedule = PassengerSchedule()
+      }
+      stay()
+    case Event(TriggerWithId(StartLegTrigger(tick, newLeg), triggerId), agentInfo) =>
+      _currentTriggerId = Some(triggerId)
+      _currentTick = Some(tick)
+      passengerSchedule.schedule.get(newLeg) match {
+        case Some(manifest) =>
+          _awaitingBoardConfirmation ++= manifest.boarders
+          manifest.riders.foreach(passenger => beamServices.vehicleRefs(passenger) ! NotifyLegStart)
+          _currentVehicle.get.ref ! UpdateTrajectory(newLeg.travelPath.toTrajectory)
+          stay()
+        case None =>
+          log.error(s"Driver ${id} did not find a manifest for BeamLeg ${_currentLeg}")
+          goto(BeamAgent.Error)
+      }
+    case Event(BoardingConfirmation(vehicleId), agentInfo) =>
+      _awaitingBoardConfirmation -= vehicleId
+      if (_awaitingBoardConfirmation.isEmpty) {
+        val theTriggerId = _currentTriggerId.get
+        _currentTriggerId = None
+        val theTick = _currentTick.get
+        _currentTick = None
+        goto(Moving) replying completed(theTriggerId, schedule[EndLegTrigger](_currentLeg.get.endTime,self))
+      } else {
+        stay()
+      }
+      val nextLeg: BeamLeg  = nextBeamLeg()
+      _currentLeg = Option(nextLeg)
+      stay() replying scheduleOne[ScheduleBeginLegTrigger](nextLeg.startTime, agent = self)
+
+  }
+
+  private def processNextLegOrCompleteMission() = {
+    val theTriggerId = _currentTriggerId.get
+    _currentTriggerId = None
+    val theTick = _currentTick.get
+    _currentTick = None
+
+    passengerSchedule.schedule.remove(passengerSchedule.schedule.firstKey)
```

Comment:
why do we do this ? Do you want to remove visited/completed stop  from schedule ?